django/db/utils.py

```diff
@@ -173,6 +173,7 @@ def ensure_defaults(self, alias):
         if conn['ENGINE'] == 'django.db.backends.' or not conn['ENGINE']:
             conn['ENGINE'] = 'django.db.backends.dummy'
         conn.setdefault('CONN_MAX_AGE', 0)
+        conn.setdefault('CONN_HEALTH_CHECK_DELAY', None)
```

Comment:
You might want to add 2 checks to the `check_settings` in the previous file (or maybe should be a system check?):

1. If `CONN_MAX_AGE == 0` then `CONN_HEALTH_CHECK_DELAY is None`.
2. If `CONN_HEALTH_CHECK_DELAY is not None` then `CONN_HEALTH_CHECK_DELAY < CONN_MAX_AGE`.